libraries/bsg_manycore_loader.cpp

```diff
@@ -179,13 +179,29 @@ static int hb_mc_loader_eva_write(const Elf32_Phdr *phdr,
                                   const hb_mc_eva_map_t *map,
                                   hb_mc_coordinate_t tile)
 {
-        hb_mc_eva_t eva = start_eva;
-        size_t off = 0, rem = sz;
+        hb_mc_npa_t start_npa;
+        size_t npa_sz = 0;
         int rc;
         char segname[64];
 
         hb_mc_loader_segment_to_string(phdr, segname, sizeof(segname));
 
+        rc = hb_mc_eva_to_npa(mc, map, &tile, &start_eva, &start_npa, &npa_sz);
+        if(rc != HB_MC_SUCCESS){
+                bsg_pr_err("%s: Failed to translate EVA into a NPA\n",
+                           __func__);
+                return rc;
+        }
+
+        if (hb_mc_manycore_npa_is_dram(mc, &start_npa)) {
+                rc = hb_mc_manycore_eva_write_dma(mc, map, &tile,
+                                                  &start_eva, data, sz);
+                if (rc == HB_MC_SUCCESS) {
+                        return rc;
+                }
```

Comment:
There's a larger space than not HB_MC_SUCCESS here. It can FAIL, or it can be UNSUPPORTED. Unsupported is fine, but FAIL is not -- the device could be in an unstable state.

Reply:
I was little unsure what are the possibilities here. Do you know where I could find the complete set of outcomes of a DMA call?

Reply:
If this documentation is insufficient, then it should be updated: 

https://github.com/bespoke-silicon-group/bsg_replicant/blob/029602cf3c6dabbf14c053334c3dca731bc51b72/libraries/bsg_manycore.h#L311

Reply:
(By me, or max, or someone else)

Reply:
That comment basically covers it... you should assume everything can also return FAIL.

Reply:
That comment basically covers it... you should assume everything can also return FAIL.